### ISSUE TITLE
pyright reports an error when function returns a type wrapped with Annotated.

### ISSUE THREAD

**Issue by marcinbarczynski** [microsoft/pyright#7865]
**Describe the bug**

Pyright reports an error when a function returns a type wrapped with `Annotated`. This is inconsistent with what `reveal_type` shows.

**Code or Screenshots**

```python
# pyright: strict
from typing import Annotated, reveal_type


class A:
    pass


def foo(cls: type[A]) -> type[A]:
    x = Annotated[cls, "Hello"]
    reveal_type(x)
    return x
```

```
$ pyright --version
pyright 1.1.361

$ pyright demo-pyright.py 
demo-pyright.py
  demo-pyright.py:11:17 - information: Type of "x" is "type[A]"
  demo-pyright.py:12:12 - error: Expression of type "type[Annotated]" is incompatible with return type "type[A]"
    Type "type[Annotated]" is incompatible with type "type[A]" (reportReturnType)
1 error, 0 warnings, 1 information 
```
